_posts/2022-04-08-reinforcement-learning.md

```diff
@@ -0,0 +1,117 @@
+- We have an agent, that can act in said environment. Agent must have some kind of input (for example camera feed), some kind of output (for example steering angle and speed), and some kind of memory, to keep track of favorable actions.
+- We also need a reward algorithm for the training. The reward algorithm decides how to interpret the environment, for example: is the car on the track? How far is it from the centerline? What's its current speed?
+- Based on the environment state, the reward algorithm will assign rewards of varying sizes, based on how well the agent is doing.
+- Reward function is only used for training the model, so once we have evaluated the model to be good enough, and we start using it, the reward function is forgotten - until the next time we want to train some models.
+- A suitable amount of chaos aka randomness is injected first into actions, so the actor will start taking random actions, to trigger the reward function evaluation. 
+- From there, the training will combine a bit of chaos, and a lot of previous learnings, and keep note of which action for which state brings the best rewards.
+- Training typically also contains evaluations, so you can focus on the best performing models.
+
+## What is AWS Deepracer?
+
+AWS Deepracer is many things. It's an AWS service you can use (for a cost). It's a physical scale model car you can drive on a physical track. It also is associated with Deepracer league, and community, where you can race your models, should you choose to do so. Races happen both virtually, and sometimes also on physical racetracks. Therefore it's also a sport, you can participate in. There are even significant rewards for the best racers, every year.
+
+It was designed to be a fun entry to machine learning and AWS services, similar to Deeplens, and Deep Composer initiatives. So it's a great way to learn about machine learning and AWS, and have some fun.
+
+Underneath the Deepracer service, there are several more basic AWS services like IAM, S3, and Sagemaker notebooks. The Robomaker service is used for the simulation environment. All these services can also be used directly for training, and there's even a local environment where you can use these as containerized services, without running anything in the cloud. Perhaps more on that in a future blog.
+
+We decided to get one physical scaled-down car, mostly for the fun factor. This is a new model, with stereo vision, and a Lidar sensor, to be able to understand its environments better than before. Having these available opens up new modes for the car. Old cars can only do time trials, but with stereo cameras and Lidar, it's also possible to do obstacle avoidance or even race with multiple cars on the track simultaneously.
+
+![AWS Deepracer car with Lidar on top](/img/reinforcement-learning-deepracer/deepracer_with_shell_side.jpg)
```

Comment:
Maybe the images could be compressed a little bit? > 2 MB per image feels a lot, I think we sould target < 1 MB per image.